test/test.js

```diff
@@ -1015,7 +1086,7 @@ describe('node-fetch', () => {
 		))
 			.to.eventually.be.fulfilled
 			.then((res) => {
-				res.body.on('error', (err) => {
+				res.body.once('error', (err) => {
```

Comment:
This is a bit weird - the error event gets emitted on the response body twice in node 14 (and 12 with this PR) - once with the expected `AbortError` and then again with `ERR_STREAM_PREMATURE_CLOSE`.

Both errors were still emitted even after changing how the `AbortError` is emitted (e.g. using `.destroy(err)` instead of `.emit('error', err)` so I'm open to suggestions on how to not need to change `.on` to `.once` here, if it's an issue.

Reply:
not sure about this, let's see what they other maintainers think, otherwise A++ pull request, thanks!

Reply:
I've refactored this away - now the implementation will check to see if the request has been aborted before emitting the `ERR_STREAM_PREMATURE_CLOSE` errors.

CI also now tests on all stable node versions 4-16 for a bit more confidence.